ilastik/applets/dataSelection/opDataSelection.py

```diff
@@ -626,14 +626,67 @@ def switch_to_scale_with_shape(self, target_shape: Dict[str, int]):
         """
-        Take the part after the last /, make it safe for use as a file name,
-        remove anything that looks like an extension ('.zarr') and replace remaining dots
-        to ensure exporting logic does not mistake them for file extensions.
+        Generate a nickname from a URL, using the pattern:
+        "last URI segment with .zarr (or .ome.zarr) plus all segments after it"
+        
+        For multiscale datasets, this creates descriptive nicknames like:
+        - "multiscale" for .../multiscale.zarr
+        - "container-multiscale" for .../container.zarr/multiscale
+        - "multiscale-scale1" for .../multiscale.zarr/scale1
+        
+        Args:
+            url: The URL to generate a nickname from
+            
+        Returns:
+            A filename-safe nickname string
         """
-        last_url_component = url.rstrip("/").rpartition("/")[2]
-        filename_safe = re.sub(r"[^a-zA-Z0-9_.-]", "_", last_url_component)
-        extensionless = os.path.splitext(filename_safe)[0]
-        return extensionless
+        # Remove trailing slashes and split into segments
+        url = url.rstrip("/")
+        
+        # Split the URL into segments (handle both file:// and https:// URLs)
+        # Remove protocol prefix for parsing
+        path_part = url
+        if "://" in url:
+            path_part = url.split("://", 1)[1]
+        
+        segments = path_part.split("/")
+        
+        # Find the last segment with .zarr or .ome.zarr extension
+        zarr_extensions = [".ome.zarr", ".zarr"]
+        zarr_index = -1
+        
+        for i in range(len(segments) - 1, -1, -1):
+            for ext in zarr_extensions:
+                if segments[i].endswith(ext):
+                    zarr_index = i
+                    break
+            if zarr_index != -1:
+                break
+        
+        # Build the nickname based on whether we found a .zarr container
+        if zarr_index == -1:
+            # No .zarr found, just use the last segment
+            nickname_parts = [segments[-1]]
+        else:
+            # Found .zarr container - collect segments from zarr_index onwards
+            nickname_parts = []
+            for i in range(zarr_index, len(segments)):
+                segment = segments[i]
+                # Strip .zarr or .ome.zarr extension from container name
+                for ext in zarr_extensions:
+                    if segment.endswith(ext):
+                        segment = segment[:-len(ext)]
+                        break
+                if segment:  # Only add non-empty segments
+                    nickname_parts.append(segment)
+        
+        # Join parts with hyphens
+        nickname = "-".join(nickname_parts) if nickname_parts else "dataset"
+        
+        # Make filename-safe (but preserve hyphens and underscores)
+        nickname = re.sub(r"[^a-zA-Z0-9_.-]", "_", nickname)
+        
+        return nickname
```

Comment:
```suggestion
        filename_safe = re.sub(r"[^a-zA-Z0-9_.-]", "_", nickname)
        
        return filename_safe
```

Style preference here, but expressive naming is better than comments